Automattic/jetpack
# Convert BackupUpgrade and BarChart to typescript

Following #23889, this PR converts the first consumer of `RecordBarMeter` to TypeScript.

#### Changes proposed in this Pull Request:

* Converts `BackupUpgrade` and `BarChart` components to TypeScript.

#### Jetpack product discussion
It was initially suggested here (https://github.com/Automattic/jetpack/pull/23889#issuecomment-1099300501) to extract these changes to a separate PR.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Run `jetpack build plugins/jetpack` and check if it's successful
* Optionally, you may also follow the testing instructions for #23374 to ensure there is no regression.
